File: easy/0944_delete_columns_to_make_sorted/solution.py
from typing import List


class Solution:
    def minDeletionSize(self, strs: List[str]) -> int:
        delete_count = 0

        cols = len(strs[0])
        rows = len(strs) - 1

        for col in range(cols):
            for row in range(rows):
                if strs[row][col] > strs[row + 1][col]:
                    delete_count += 1
                    break

        return delete_count

File: easy/0944_delete_columns_to_make_sorted/solution.cpp
using namespace std;

#include <string>
#include <vector>

class Solution {
public:
  int minDeletionSize(vector<string> &strs) {
    int deleteCount = 0;

    const int cols = strs[0].size();
    const int rows = strs.size() - 1;

    for (int col = 0; col < cols; ++col) {
      for (int row = 0; row < rows; ++row) {
        if (strs[row][col] > strs[row + 1][col]) {
          ++deleteCount;
          break;
        }
      }
    }

    return deleteCount;
  }
};
